sonar-python-plugin/src/main/java/org/sonar/plugins/python/SensorTelemetryStorage.java

```diff
@@ -42,8 +43,19 @@ public Map<String, String> data() {
   }
 
   public void send(SensorContext sensorContext) {
-    data.forEach(sensorContext::addTelemetryProperty);
     data.forEach((k, v) -> LOG.info("Metrics property: {}={}", k, v));
+    var apiVersion = sensorContext.runtime().getApiVersion();
+    if (apiVersion.isGreaterThanOrEqual(Version.create(10, 9))) {
+      // This try/catch block should be useless, as in the worst case it should be a no-op depending on the SensorContext implementation
+      // It exists to be extra sure for the LTA
```

Comment:
If you want to be super-sure that no exception will be thrown up then probably the `try` statement should wrap entire method like:
```java
try {
  data.forEach((k, v) -> LOG.info("Metrics property: {}={}", k, v));
  var apiVersion = sensorContext.runtime().getApiVersion();
  if (apiVersion.isGreaterThanOrEqual(Version.create(10, 9))) {
    data.forEach(sensorContext::addTelemetryProperty);
  } else  {
    LOG.info("Skipping sending metrics because the plugin API version is {}", apiVersion);
  }
} catch (Exception e) {
  LOG.error("Failed to send metrics", e);
}
```